### PR TITLE
remove_edges_under_consolidation

### PR DESCRIPTION
**Pull request by rkvinoth**
Solution to https://github.com/cubewise-code/tm1py/issues/322#issue-675997568
Any ideas on refactoring?

### CODE REVIEW

File: TM1py/Services/HierarchyService.py [hunk 1]
```diff
@@ -211,6 +211,30 @@ def remove_all_edges(self, dimension_name: str, hierarchy_name: str = None, **kw
...
+        elements_under_consolidations = element_service.get_members_under_consolidation(dimension_name, hierarchy_name,
+                                                                                        consolidation_element)
+        elements_under_consolidations.append(consolidation_element)
+        remove_edges = []
+        for element, _ in hierarchy.edges.items():
+            if element[0] in elements_under_consolidations and element[1] in elements_under_consolidations:
+                remove_edges.append(element)
+        hierarchy.remove_edges(remove_edges)
+
+        url = format_url("/api/v1/Dimensions('{}')/Hierarchies('{}')", dimension_name, hierarchy_name)
```
Comment (MariusWirtz) [inline TM1py/Services/HierarchyService.py:234]: I think you can just use the `self.update` function here instead

File: TM1py/Services/HierarchyService.py [hunk 2]
```diff
@@ -211,6 +211,30 @@ def remove_all_edges(self, dimension_name: str, hierarchy_name: str = None, **kw
...
+        :return: response
+        """
+        hierarchy = self.get(dimension_name, hierarchy_name)
+        from TM1py.Services import ElementService
+        element_service = ElementService(self._rest)
+        elements_under_consolidations = element_service.get_members_under_consolidation(dimension_name, hierarchy_name,
+                                                                                        consolidation_element)
+        elements_under_consolidations.append(consolidation_element)
+        remove_edges = []
+        for element, _ in hierarchy.edges.items():
```
Comment (MariusWirtz) [inline TM1py/Services/HierarchyService.py:229]: I think it's better to write 
`for edge in hierarchy.edges:`
here. Or:
I think it's better to write 
`for (parent, component) in hierarchy.edges:`
here.

**Comment by MariusWirtz** [comment]
Thanks for solving #322

I don't like that we have to retrieve the full hierarchy and send it back. It's going to be slow on big dimensions.
For the moment I see no better approach than the one you implemented though.

Once TM1 supports multi delete operations on collections, we could add a `delete_edges` function to this service and simplify (and speed up) this function.

File: TM1py/Services/HierarchyService.py (hunk 2 above)
Comment (rkvinoth) [reply TM1py/Services/HierarchyService.py:229]: nice..

File: TM1py/Services/HierarchyService.py (hunk 1 above)
Comment (rkvinoth) [reply TM1py/Services/HierarchyService.py:234]: agreed..

**Comment by rkvinoth** [comment]
That's right.. even I was expecting a collection delete. 
Anyway let's use the available resources for now.

**Pull request merged**